File: 974_subarray_sums_divisible_by_k/subarraysDivByK.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        count = [1] + [0] * k
        s = 0
        res = 0

        for num in nums:
            s = (s + num) % k
            res += count[s]
            count[s] += 1


        return res

File: 974_subarray_sums_divisible_by_k/subarraysDivByK.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        int[] count = new int[k];
        count[0] = 1;

        int res = 0;
        int sum = 0;

        for (int num : nums) {
            sum = (sum + num) % k;
            if (sum < 0) sum += k;
            res += count[sum]++;
        }

        return res;
    }
}
